src/Npgsql/NpgsqlCommand.cs

```diff
@@ -615,14 +615,14 @@ async Task PrepareLong(CancellationToken cancellationToken)
                         {
                             if (pStatement.StatementBeingReplaced != null)
                             {
-                                Expect<CloseCompletedMessage>(await connector.ReadMessage(async, cancellationToken), connector);
+                                Expect<CloseCompletedMessage>(await connector.ReadMessageWithCancellation(async, cancellationToken), connector);
```

Comment:
Shouldn't ReadMessage just be with cancellation by default? After all, the only case where we don't want cancellation when reading a message is for Wait. We also already have ReadMessageWithNotifications which Wait goes through (to avoid breaking etc.), so should we just make that disable cancellation instead?

Reply:
There is also COPY and authentication. But I don't mind changing this.

Reply:
For COPY export, when reading a message we can do PG cancellation, no? For auth you're right.

In any case, the main flow in Npgsql is command execution, so the cases where we read messages without cancellation are more the edge case... I'd flip this.

Reply:
For some reason, I thought every COPY operation has a different mechanism for the cancellation. But no, for the export it's the same as for the main flow.